stages/text-extraction/src/main/java/edu/kit/kastel/mcse/ardoco/core/textextraction/OriginalTextStateStrategy.java

```diff
@@ -2,43 +2,35 @@
 package edu.kit.kastel.mcse.ardoco.core.textextraction;
 
 import java.util.Arrays;
-import java.util.Objects;
-import java.util.function.BiPredicate;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
 import org.eclipse.collections.api.factory.Lists;
-import org.eclipse.collections.api.factory.Maps;
-import org.eclipse.collections.api.factory.Sets;
+import org.eclipse.collections.api.factory.SortedMaps;
+import org.eclipse.collections.api.factory.SortedSets;
 import org.eclipse.collections.api.list.ImmutableList;
 import org.eclipse.collections.api.list.MutableList;
-import org.eclipse.collections.api.map.MutableMap;
-import org.eclipse.collections.api.set.MutableSet;
+import org.eclipse.collections.api.map.sorted.MutableSortedMap;
+import org.eclipse.collections.api.set.sorted.MutableSortedSet;
 
 import edu.kit.kastel.mcse.ardoco.core.api.text.Word;
 import edu.kit.kastel.mcse.ardoco.core.api.textextraction.MappingKind;
 import edu.kit.kastel.mcse.ardoco.core.api.textextraction.NounMapping;
-import edu.kit.kastel.mcse.ardoco.core.common.util.ElementWrapper;
 import edu.kit.kastel.mcse.ardoco.core.common.util.SimilarityUtils;
 import edu.kit.kastel.mcse.ardoco.core.data.Confidence;
 import edu.kit.kastel.mcse.ardoco.core.pipeline.agent.Claimant;
 
 public class OriginalTextStateStrategy extends DefaultTextStateStrategy {
 
-    private static final Function<NounMapping, Integer> NOUN_MAPPING_HASH = nm -> Objects.hash(nm.getReferenceWords().toSet(), nm.getWords());
-    private static final BiPredicate<NounMapping, NounMapping> NOUN_MAPPING_EQUALS = (nm1, nm2) -> Objects.equals(nm1.getReferenceWords().toSet(), nm2
-            .getReferenceWords()
-            .toSet()) && Objects.equals(nm1.getWords(), nm2.getWords());
-
     OriginalTextStateStrategy(TextStateImpl textState) {
         super.setTextState(textState);
     }
 
     @Override
     public NounMapping addOrExtendNounMapping(Word word, MappingKind kind, Claimant claimant, double probability, ImmutableList<String> surfaceForms) {
 
-        NounMapping disposableNounMapping = new NounMappingImpl(System.currentTimeMillis(), Sets.immutable.with(word), kind, claimant, probability,
-                Lists.immutable.with(word), surfaceForms);
+        NounMapping disposableNounMapping = new NounMappingImpl(NounMappingImpl.getNextCreationTime(), SortedSets.immutable.with(word), kind, claimant,
```

Comment:
Isn't it better to offer a second constructor (or change the existing one) that has the first parameter (creationTime) removed? It is odd when a class calls the constructor and needs to access a date from the class it tries to construct.